### PR TITLE
Fix issue with file in path

### LINKED ISSUE #80: Internal error on paths like '/static/a/abc' if 'a' is a file
### Problem
Accessing a path like `static/a/abc` will throw an error and respond with status code 500 if `a` is a file. This is because this generates an `ENOTDIR` error, instead of `ENOENT`.

We discovered this since we serve images at `static/*.jpg` and paths like `static/*.jpg/abc` returned 500 instead of 404. 

### Expectation
Should respond with status code 404 and show the 404 page.

### Solution
In the code you check for`if (err.code !== 'ENOENT')`. Adding `&& err.code !== 'ENOTDIR'` for example here makes it correctly respond with 404 instead of 500:

https://github.com/zeit/serve-handler/blob/d1368bf5ece565b27850ceaab2758a5ae0a4579d/src/index.js#L611

### PR DESCRIPTION
**Pull request by AxlLind**
Fixes error described in #80. See issue for problem description.

Resolves #80

### CODE REVIEW

**Comment by JohanBengtsson** [comment]
Nice PR! Would love if this could be merged!

File: src/index.js [hunk 1]
```diff
@@ -582,7 +582,7 @@ module.exports = async (request, response, config = {}, methods = {}) => {
 		try {
 			stats = await handlers.stat(absolutePath);
 		} catch (err) {
-			if (err.code !== 'ENOENT') {
+			if (err.code !== 'ENOENT' && err.code !== 'ENOTDIR') {
```
Comment (r0mflip) [inline src/index.js]: Wouldn't this cause problems with rewrites? When there directory might not exist but a rewrite for it does?

Comment (AxlLind) [reply src/index.js]: Not sure I follow, could you provide an example of what you mean?

Comment (r0mflip) [reply src/index.js]: Thought there would be problems with such config where `static/b` doesn't exist

``` js
await handler(req, res, {
  'rewrites': [
    {source: 'static/b/**', destination: '/static/index.html'}
  ],
});
```

But can confirm, no such problem, my bad.

**Comment by AxlLind** [comment]
@leo You seem to be the maintainer to pinging you.

We get errors in our production logs from this. Would be really great if we could merge this, seems like a safe deploy to me.

**Comment by JohanBengtsson** [comment]
@leo  thanks!

**Pull request merged**